src/rgw/rgw_rados.cc

```diff
@@ -1881,30 +1879,14 @@ int RGWRados::Bucket::List::list_objects_ordered(int64_t max_p,
 
       result->emplace_back(std::move(entry));
       count++;
-    }
-
-    if (!params.delim.empty()) {
-      int marker_delim_pos = cur_marker.name.find(params.delim, cur_prefix.size());
```

Comment:
am i correct that we're removing this filtering from radosgw itself, on the assumption that the osd is doing it for us? i worry about cases where radosgw is upgraded to include this change, but the osds are not - which would leave these common entries unfiltered

to avoid that, i feel like we'd need to preserve this filtering logic in radosgw for another couple releases. :(

we'd still get the correct result if filtering happens in both locations, right?

Reply:
@cbodley I assume we do not need to execute the logic in radosgw when the OSD -does- the filtering...

Reply:
Then do we need a way to determine if the filtering did occur.... I can bump up the struct version # of rgw_cls_list_ret. I'd also add a boolean, such as cls_decoded. That boolean would never be transmitted. However the version of the structure would determine if it was true or false. If it was false, then the filtering would happen on the rgw side, but not if it's true.

Reply:
right, rgw wouldn't know unless we extend the cls protocol. but even if the osd does this filtering, duplicating it in rgw should essentially be a noop right? if that's the case, i'd prefer to leave it as is, rather than complicating it further by making it conditional

Reply:
Because I already did the work in cleaning up the filtering, I'd rather have a conditional. There are comments in the code that once we reach version 16 of Ceph, we can remove the compatibility code.

I ran the s3-tests in 3 combinations and got the same result each time:
- new rgw, new osd : FAILED (SKIP=5, errors=22, failures=25)
- new rgw, old osd: FAILED (SKIP=5, errors=22, failures=25)
- old rgw, old osd: FAILED (SKIP=5, errors=22, failures=25)

So it all seems to work. The above includes 27 boto 3 tests with the prefixes "s3tests_boto3.functional.test_s3.test_bucket_list_delimiter*" and "s3tests_boto3.functional.test_s3.test_bucket_listv2_delimiter*".